docs/user/technical-reference/external-authorizer.md

```diff
@@ -0,0 +1,129 @@
+spec:
+  action: ALLOW
+  rules:
+  - to:
+      - operation:
+        paths:
+          - /headers
+    when:
+      - key: request.auth.claims[some_claim]
+        values:
+          - some_value
+```
+
+- and an additional `RequestAuthentication` that makes sure Istio recognizes the issuer:
+```yaml
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: httpbin
+spec:
+  jwtRules:
+  - issuer: https://example.com
+    jwksUri: https://example.com/.well-known/jwks.json
+```
+
+**Decision**
+We decided to enforce that the user **MUST** have **ONE**, and **ONLY ONE** `accessStrategy` per every entry in `spec.rules`. As so, we don't support mixing up `extAuth` and `jwt`.
+Instead, we would like to allow configuration in scope of the `extAuth` strategy, and creating a `ALLOW` AuthorizationPolicy based on that configuration.
+In result, the proposed API would look as follows:
+
+```yaml
+accessStrategy: # Validation: there needs to be one access strategy, and only one
+  extAuth:
+    name: oauth2-proxy
+    restrictions: # Feedback about name and structure appreciated
+      # Will most likely have the same structure as in `jwt` access strategy
+      authentications:
+        - issuer: https://example.com
+          jwksUri: https://example.com/.well-known/jwks.json            
+      authorizations:
+        - audiences: ["app1"]
+```
+
+### Should we support multiple external authorizers?
+
+We need to consider whether a configuration that will use multiple external authorizers on one path is valuable. Technically, this is possible to do, as all CUSTOM policies will need to result in `allow` response for the request to be allowed.
+
+**Decision**
+We decided to not support multiple external authorizers. Supporting multiple external authorizers will introduce additional complexity in the API, that might lead to unexpected behaviour or/and confuse the user.
+If there is a use case for configuring multiple external authorizers, there is still a posibillity of creating the required `AuthorizationPolicies` themselves.
+
+## API Proposal
+
+We have discussed how the API would look like in context of support in future `v1beta2/v1` versions. We decided that `accessStrategy` will hold a single entry, either `extAuth`, `jwt` or `noAuth`. The users **MUST** define **ONE**, and **ONLY ONE** access strategy in every `rule` in `spec.rules`.
+
+A sample using the proposed API would look as follows:
+
+```yaml
+apiVersion: gateway.kyma-project.io/v1beta1
+kind: APIRule
+metadata:
+  name: service-secured
+spec:
+  gateway: kyma-system/kyma-gateway
+  host: httpbin
+  service:
+    name: httpbin
+    port: 8000
+  rules:
+    - path: /headers
+      methods: ["GET"]
+      mutators:
+        - ***
+      accessStrategy: # Validation: there needs to be one access strategy, and only one
+        extAuth:
+          name: oauth2-proxy # Validation: Check if there is that authorizer in Istio mesh config
+          restrictions:
```

Comment:
Do we need this restrictions field? Maybe it's better to have it on the same level as in the jwt strategy.

Reply:
Yeah, maybe it's better from API perspective
Just thought that this will make it easier to support this, as we can make a composition from the structure that will represent `extAuth`. 
Like this:

```go
type ExtAuthHandler struct{
  Restrictions JwtConfig
}
```

Though I think the same solution would be using unnamed field:
```go
type ExtAuthHandler struct{
  JwtConfig
}
```
I'm only not sure whether kubebuilder generator will be fine with that 😠

Reply:
OK, I went through with deleting this field